openmetadata-ui/src/main/resources/ui/src/components/common/Form/JSONSchema/JsonSchemaWidgets/PasswordWidget.tsx

```diff
@@ -25,28 +37,71 @@ const PasswordWidget: FC<WidgetProps> = (props) => {
     }
   }, [props.value]);
 
-  if (props.schema.uiFieldType === 'file') {
-    return <FileUploadWidget {...props} />;
-  }
+  const getPasswordInput = useCallback(
+    (disabled?: boolean) => (
+      <Input.Password
+        autoComplete="off"
+        autoFocus={props.autofocus}
+        data-testid={`password-input-widget-${props.id}`}
+        disabled={disabled || props.disabled}
+        id={props.id}
+        name={props.name}
+        placeholder={props.placeholder}
+        readOnly={props.readonly}
+        required={props.required}
+        value={passwordWidgetValue}
+        onBlur={() => props.onBlur(props.id, props.value)}
+        onChange={(e) => props.onChange(e.target.value)}
+        onFocus={() => props.onFocus(props.id, props.value)}
+      />
+    ),
+    [props]
+  );
 
-  const { onFocus, onBlur, onChange, ...rest } = props;
+  const onRadioChange = (e: RadioChangeEvent) => {
+    setInputType(e.target.value);
+  };
 
   return (
-    <Input.Password
-      autoComplete="off"
-      autoFocus={rest.autofocus}
-      data-testid="password-input-widget"
-      disabled={rest.disabled}
-      id={rest.id}
-      name={rest.name}
-      placeholder={rest.placeholder}
-      readOnly={rest.readonly}
-      required={rest.required}
-      value={passwordWidgetValue}
-      onBlur={() => onBlur(rest.id, rest.value)}
-      onChange={(e) => onChange(e.target.value)}
-      onFocus={() => onFocus(rest.id, rest.value)}
-    />
+    <>
+      {isInputTypeFile ? (
```

Comment:
Thinking out loud, but don't you think uiType file will always be rendering this widget now?
Have you check that actually we needed this behvaiour for all the places where we have uiType File? 
I may call it fileOrInput to be more clear

Reply:
We are currently using this only for the certificate inputs now. But I agree with you here. We should make that clear in the UI type if we want an alternative. Thanks.